packages/lib/JoplinDatabase.ts

```diff
@@ -874,6 +874,16 @@ export default class JoplinDatabase extends Database {
 				queries.push('ALTER TABLE folders ADD COLUMN share_id TEXT NOT NULL DEFAULT ""');
 				queries.push('ALTER TABLE notes ADD COLUMN share_id TEXT NOT NULL DEFAULT ""');
 				queries.push('ALTER TABLE resources ADD COLUMN share_id TEXT NOT NULL DEFAULT ""');
+
+				queries.push('DROP VIEW tags_with_note_count');
+				queries.push(`CREATE VIEW tags_with_note_count AS 
+						SELECT tags.id as id, tags.title as title, tags.created_time as created_time, tags.updated_time as updated_time, COUNT(notes.id) as note_count, 
+							SUM(CASE WHEN notes.todo_completed > 0 THEN 1 ELSE 0 END) AS todo_completed_count
+						FROM tags 
+							LEFT JOIN note_tags nt on nt.tag_id = tags.id 
+							LEFT JOIN notes on notes.id = nt.note_id 
+						WHERE notes.id IS NOT NULL 
+						GROUP BY tags.id`);
```

Comment:
We can't add to a migration because it means it won't run for those who are, for example, already at revision 36. So every time you need to add something to the db, you need to create a new migration.

By the way, what will be the performance of dropping and recreating this table? Will it result in a white screen on mobile while it's upgrading?

Reply:
This happened because we both pushed a 36 and then I didn't raise when resolving the conflict.
So it is best to create a separate DB revision for each PR?
Even if there are in the worst case then x revisions for a new released version?

I just see there is already a 37, even if nothing was done in this one I have to create a 38?
https://github.com/laurent22/joplin/blob/dc2cdb7d3ae751e720c1bb40b53dc048b254862d/packages/lib/JoplinDatabase.ts#L346

>By the way, what will be the performance of dropping and recreating this table? 

Since it is only a view, this should actually have no performance impact. 
But I'm willing to try on my phone with a few thousand notes and tags, but this takes some time for the sync :)

Reply:
Indeed it looks like I've accidentally added 37 even though nothing was done. So the next one will have to be 38, otherwise existing clients won't pick up the migration.

And yes, it should be a separate revision for each pull request and, if there are conflicts, we increase the number.